### DISCUSSION TITLE
How can I use the docker setup behind an nginx proxy? I already have nginx on my host system and docker seems to want to install its own which conflicts with port 80

### DISCUSSION THREAD

**Discussion by ralyodio** [langgenius/dify#12229] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

I'm trying to setup this to run on https://*.*.dify.profullstack.com which is behind the HOST nginx server already setup and hosting numerous other services.

However when I run the `docker compose up -d` i get the following error:

```
 ⠿ Container docker-nginx-1           Starting                                                                                                                                                          16.9s
Error response from daemon: driver failed programming external connectivity on endpoint docker-nginx-1 (08b7f71f798954f9c5a5d5fb81c3f8caa5c31d518b0499175437d79e97c32f32): failed to bind port 0.0.0.0:80/tcp: Error starting userland proxy: listen tcp4 0.0.0.0:80: bind: address already in use
```

I need to make this work on a non-443 or non-80 port so I can proxy it through from my host nginx to the localhost:whatever port.

**Comment by HenryXiaoYang** [comment]
I am having this same issue! Need some support.

**Comment by mounta11n** [comment]
the error says: ` failed to bind port 0.0.0.0:80/tcp: Error starting userland proxy: listen tcp4 0.0.0.0:80: bind: address already in use`

because your nginx is listening on port 80.
just change the port in your docker-compose.yaml from `80:80` to `127.0.0.1:80:80` or anything else. then let nginx just point to the docker-container-ip:80 or simply to the hostname of the container.

btw, it is from a security view very crucial to **not expose** your container ports to 0.0.0.0, since this is the whole point with reverse proxies like nginx: you want to hide (127.0.0.1) everything behind your reverse proxy, so that nobody can access your containers directly from the internet (which would be the case with 0.0.0.0), but only make requests to the reverse proxy/nginx

**Reply by mounta11n** [reply]
> from 80:80 to 127.0.0.1:80:80 or anything else

just to clarify, with anything else i mean -> `127.0.0.7:ANY-PORT:80` e.g. `127.0.0.7:55234:80`

**Comment by mu-dan** [comment]
可以先把目前非docker部署的nginx进程杀死，然后部署1panel（类似宝塔面板，但是我觉得比宝塔好用），在1panel容器里面对dify所需要的nginx端口进行修改，最后再启用外部的nginx进程。

**Reply by mu-dan** [reply]
![image](https://github.com/user-attachments/assets/8aa3d15e-31fc-4fc5-9c64-3f7b4410bc23)
![image](https://github.com/user-attachments/assets/ee3b01b6-d110-46b4-8f33-88e9087c8467)
然后点击确认就行了，如果报错就把nginx容器停止再进行修改
![image](https://github.com/user-attachments/assets/6758127a-230e-4e91-bc8b-e8658adc27ea)
